### PR TITLE
Feature/ci v2 protocol

### PR DESCRIPTION
**Pull request by ori-gold-px**
* added CI V2 protocol

### CODE REVIEW

File: src/CredentialsIntelligence/Protocol/V2CredentialsIntelligenceProtocol.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+        if ($domain === "@gmail.com") {
+            $normalizedUsername = str_replace(".", "", $normalizedUsername);
+        }
+
+        return $normalizedUsername . $domain;
+    }
+
+    private function hashPassword($salt, $password) {
+        $hashedPassword = PerimeterxUtils::sha256($password);
```
Comment (chen-zimmer-px) [inline src/CredentialsIntelligence/Protocol/V2CredentialsIntelligenceProtocol.php:40]: I suggest creating a new function that gets the required hash algorithm along with the function.
Then, passing the hash algorithm type (that will be a constant variable) from `hashPassword` function, or create dedicated SHA_256 function that uses the former function.

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Added
```
Comment (chen-zimmer-px) [inline CHANGELOG.md:10]: Please add some unit tests.

**Review by chen-zimmer-px — CHANGES_REQUESTED** [review]
Please fix the comment and add some unit tests.

File: CHANGELOG.md (hunk 2 above)
Comment (ori-gold-px) [reply CHANGELOG.md:10]: Done!

Comment (ori-gold-px) [reply CHANGELOG.md:10]: Done!

**Review by AsafAklerPX — APPROVED** [review]
Approved!

**Pull request merged**